### DISCUSSION TITLE
Pull data from Quickbooks Online

### DISCUSSION THREAD

**Discussion by rmutili** [Budibase/budibase#2650] [Help]
Hi guys,

Am trying to automate creation of custom made receipt in a pdf format once a payment in QuickBooks. I would like to pull the data from quickbooks payment table, select and filter some fields to a table in budibase then select the items I need receipt to be created. Once I select the items I need to create (need to be able to select many of them at a go) I will need to have "Send Email" button which will send separate pdfs from each row to an email address within the row using a predefined pdf template.

Can someone help me on how to achieve the above automation with specific steps and screenshots if possible.

Kind Regards,

Raphael

**Comment by mjashanks** [comment]
Hi Raphael. Unfortunately, this is not possible. The missing piece is the ability to run an action **for each** row.

We actually need this in 2 places for your automation
1. When pulling payments from Quickbooks, we need to create a row, for each payment
2. After selecting items, we need send an email or each item.

I think we really need this feature but have not scheduled it yet.

**Comment by rmutili** [comment]
Thanks @mjashanks, exciting to be part of the journey...
